### PR TITLE
[GLT-3314] Filter out certain tube numbers from PRECISE Case Report

### PR DESCRIPTION
**Pull request by avarsava**
They are outliers adn should not be counted

### CODE REVIEW

File: src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseCaseReport.java [hunk 1]
```diff
@@ -151,6 +155,9 @@ protected void collectData(PineryClient pinery) throws HttpResponseException, IO
                 allPreciseSamplesByIdentity
                     .get(identity.getId()) // Get children of this identity
                     .stream()
+                    .filter(s -> 06 != SampleUtils.getTubeNumber(s.getName())) // See GLT-3314
+                    .filter(s -> 17 != SampleUtils.getTubeNumber(s.getName()))
+                    .filter(s -> 31 != SampleUtils.getTubeNumber(s.getName()))
```
Comment (djcooke) [inline src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseCaseReport.java]: Add a static `Predicate` that excludes based on all 3 tube numbers and reference that in all 4 places this is done, so the tube numbers are defined in a single place. While it probably won't do any harm in this instance, the `6` is specified in octal and shouldn't be

Comment (avarsava) [reply src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseCaseReport.java]: Fixed one and forgot to fix the others, so putting them all in one place is probably a good idea

Comment (avarsava) [reply src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseCaseReport.java]: Done

**Review by djcooke — APPROVED** [review]

**Review by apmasell — APPROVED** [review]

**Comment by avarsava** [comment]
(Waiting for Ilinca to approve results)

File: src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseCaseReport.java [hunk 2]
```diff
@@ -169,7 +169,8 @@ protected void collectData(PineryClient pinery) throws HttpResponseException, IO
                                 || "POS"
                                     .equals(
                                         getUpstreamAttribute(
-                                            ATTR_GROUP_ID, s, allPreciseSamplesById)))
+                                            ATTR_GROUP_ID, s, allPreciseSamplesById))
+                                || null == getAttribute(ATTR_GROUP_ID, s))
```
Comment (djcooke) [inline src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseCaseReport.java]: This is only checking upstream (parent samples) for "POS" and only checking the current sample for null? Probably want to check both in both cases?

Comment (avarsava) [reply src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseCaseReport.java]: Ilinca says the current counts are correct, and adding the upstream null check does change the count :shrug:

Comment (avarsava) [reply src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseCaseReport.java]: (It does also check the current sample for POS too though)

Comment (djcooke) [reply src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseCaseReport.java]: Ah, I missed the other check for "POS" on current sample.

`getUpstreamAttribute` should be checked for null too though. There could be a tissue with group ID "NEG" with a child slide not specifying a group ID. In this case, the tissue's "NEG" group ID should apply to the child, but the report will count it as "POS" instead. While there may not be any samples matching this pattern right now, there could be in the future before this report is rerun.

**Review by apmasell — APPROVED** [review]

File: src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseCaseReport.java (hunk 2 above)
Comment (avarsava) [reply src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseCaseReport.java]: sorry, I'd misunderstood how this check should be implemented. Fixed and verified no data change

**Review by djcooke — APPROVED** [review]

**Pull request merged**